core/Rename.cpp

```diff
@@ -272,33 +268,52 @@ namespace olympia
 
                 const auto & srcs = renaming_inst->getSourceOpInfoList();
                 const auto & dests = renaming_inst->getDestOpInfoList();
-
                 for(const auto & src : srcs)
                 {
-                    const auto rf  = olympia::coreutils::determineRegisterFile(src);
-                    const auto num = src.field_value;
-                    auto & bitmask = renaming_inst->getSrcRegisterBitMask(rf);
-                    const uint32_t prf = map_table_[rf][num];
-                    reference_counter_[rf][prf]++;
-                    renaming_inst->getRenameData().setSource({prf, rf});
-                    bitmask.set(prf);
-
-                    ILOG("\tsetup source register bit mask "
-                         << sparta::printBitSet(bitmask)
-                         << " for '" << rf << "' scoreboard");
+                    if(renaming_inst->isStoreInst() && dests.size() == 0 && (src.field_id == mavis::InstMetaData::OperandFieldID::RS2)){
+                        // for store data operand we store the data reg information differently
+                        const auto rf  = olympia::coreutils::determineRegisterFile(src);
+                        const auto num = src.field_value;
+                        auto & bitmask = renaming_inst->getDataRegisterBitMask(rf);
+                        const uint32_t prf = map_table_[rf][num];
+                        reference_counter_[rf][prf]++;
+                        renaming_inst->getRenameData().setDataReg({prf, rf});
+                        bitmask.set(prf);
 
-                    if(renaming_inst->isStoreInst() && srcs.size() < 2 && dests.size() == 0){
-                        // assigning a register from PRF for data register
-                        auto & addr_bitmask = renaming_inst->getAddrRegisterBitMask();
-                        core_types::RegFile rf = core_types::RF_INTEGER; // all address operations are integer
+                        ILOG("\tsetup store data register bit mask "
+                            << sparta::printBitSet(bitmask)
+                            << " for '" << rf << "' scoreboard");
+                    }
+                    else if(renaming_inst->isStoreInst() && dests.size() == 0){
```

Comment:
This statement can be combined with the previous `if` statement.  Here's the logic:

```
for (src : src_list)
{
    if( store instruction ) {
       if( src.pos == RS2 ) {
            setDataReg(src);
       }
        else {
            setSource(src);
        }
    }
}
```

Reply:
Implemented in [e7fa84c](https://github.com/riscv-software-src/riscv-perf-model/pull/70/commits/e7fa84ca02af05a0422db09893bfd2815f6b2e11)